scripts/init.groovy.d/aa_configure_credentials.groovy

```diff
@@ -0,0 +1,370 @@
+// output init hook stage
+println '### JENKINS AUTOCONFIG: CREDENTIALS ###'
+
+/**
+  * Check if Jenkins should not configure itself and skip.
+  */
+
+Boolean skipConfiguringJenkins() {
+    !(System.env?.ENVIRONMENT in ['dev', 'staging']) ||
+        (new File("${Jenkins.instance.rootDir}/autoConfigComplete").exists())
+}
+if(skipConfiguringJenkins()) {
+    println 'Skipping Jenkins auto-configuration'
+    return
+}
+
+
+/**
+  * From this point onward Jenkins should configure itself.
+  */
+import jenkins.*
+import jenkins.model.*
+import hudson.*
+import hudson.model.*
```

Comment:
I left this similar feedback elsewhere, but — if possible — I'd recommend removing wildcard imports and importing only the required classes.